tests/scripts/thread-cert/node.py

```diff
@@ -637,7 +575,10 @@ def get_eidcaches(self):
         self.send_command('eidcache')
 
         while True:
-            i = self._expect([r'([a-fA-F0-9\:]+) ([a-fA-F0-9]+)\r?\n', 'Done'])
+            i = self._expect([
+                _EXPECT_LINE_START + r'([a-fA-F0-9\:]+) ([a-fA-F0-9]+)' +
+                _EXPECT_LINE_END, 'Done'
+            ])
```

Comment:
What about something like below
```suggestion
            i = self._expect_results(r'(([a-fA-F0-9\:]+) ([a-fA-F0-9]+)|Done)')
```

Or if we just stop matching  when we found `Done` or `Error: ...`, we can even simplify this to 
`i = self._expect_results(r'([a-fA-F0-9\:]+) ([a-fA-F0-9]+)')`

Thoughts?